### PR TITLE
Update base docker image to ubuntu 22.04, optimize container size.

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR update docker builds steep to use ubuntu 22.04 and python 3.9

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: dockerfile [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 # install graphical libraries used by qt and vispy
-RUN apt-get install -qqy mesa-utils libgl1-mesa-glx  libglib2.0-0
-RUN apt-get install -qqy libfontconfig1 libxrender1 libdbus-1-3 libxkbcommon-x11-0 libxi6
-RUN apt-get install -qqy libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0
-RUN apt-get install -qqy libxcb-xinerama0 libxcb-xinput0 libxcb-xfixes0 libxcb-shape0
+RUN apt-get install -qqy mesa-utils libgl1-mesa-glx  libglib2.0-0 \
+    libfontconfig1 libxrender1 libdbus-1-3 libxkbcommon-x11-0 libxi6 \
+    libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 \
+    libxcb-xinerama0 libxcb-xinput0 libxcb-xfixes0 libxcb-shape0 \
+    && apt-get clean
```
Comment (Czaki) [inline dockerfile]: clean cache after install reduce container size

File: dockerfile [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
 
 # install napari release version
-RUN pip3 install napari[all]
+RUN pip3 install napari[all] scikit-image && pip3 cache purge
 
-# install scikit image for examples
-RUN pip3 install scikit-image
```
Comment (Czaki) [inline dockerfile]: scikit-image was added as dependency to napari in #3867

**Comment by Czaki** [comment]
@truatpasteurdotfr may you hint to me why singularity steep fail?

**Comment by truatpasteurdotfr** [comment]
` E: Package 'python3.8' has no installation candidate` for ubuntu/jammy (as :latest has shifted). imho...
May pinning to ubuntu LTS 20.04 would be better

**Comment by Czaki** [comment]
> ` E: Package 'python3.8' has no installation candidate` for ubuntu/jammy (as :latest has shifted). imho... May pinning to ubuntu LTS 20.04 would be better

you point error on main branch. 

I enable build in this PR and docker build pass (I pin Dcokerfile version to 22.04 and change python to 3.9) but singularity fail with:
```
 tox.ini
INFO:    Starting build...
FATAL:   While performing build: conveyor failed to get: unable to retrieve auth token: invalid username/password: unauthorized
Error: Process completed with exit code 255.
```

https://github.com/napari/napari/runs/6588822234?check_suite_focus=true#step:5:39

**Comment by truatpasteurdotfr** [comment]
Hi, the artefacts produced by docker must be accessible...  I think that you need to make public https://github.com/orgs/napari/packages?repo_name=napari

Cheers

Tru

**Comment by sofroniewn** [comment]
> Hi, the artefacts produced by docker must be accessible... I think that you need to make public https://github.com/orgs/napari/packages?repo_name=napari

I have just made them public (@jni let me know if there was any reason we wanted them private)

**Comment by Czaki** [comment]
For reduce container size and reduce probability of false negative result we could split build containers into two jobs. The first one will install all necessary libraries from ubuntu repository and could be run on weekly/monthly schedule and the second job will only install python libraries

File: dockerfile (hunk 2 above)
Comment (brisvag) [reply dockerfile]: So why is it needed above in the pip call? Isn't it handled already by `napari[all]`?

Comment (Czaki) [reply dockerfile]: this is remove of this lines

File: dockerfile [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
-RUN apt-get install -qqy libxcb-xinerama0 libxcb-xinput0 libxcb-xfixes0 libxcb-shape0
+RUN apt-get install -qqy mesa-utils libgl1-mesa-glx  libglib2.0-0 \
+    libfontconfig1 libxrender1 libdbus-1-3 libxkbcommon-x11-0 libxi6 \
+    libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 \
+    libxcb-xinerama0 libxcb-xinput0 libxcb-xfixes0 libxcb-shape0 \
+    && apt-get clean
 
 # install napari release version
-RUN pip3 install napari[all]
+RUN pip3 install napari[all] scikit-image && pip3 cache purge
```
Comment (brisvag) [inline dockerfile]: I meant this where `scikit-image` was added.

File: dockerfile (hunk 2 above)
Comment (Czaki) [reply dockerfile]: When Dockerfile was created the scikit-image was not a napari dependency.

**Comment by sofroniewn** [comment]
@Czaki do you need further support here? Anything we can do to help get this merged?

**Comment by Czaki** [comment]
> Anything we can do to help get this merged?

Zrzut ekranu z 2022-07-19 09-39-11

**Review by sofroniewn — APPROVED** [review]
LGTM!

**Pull request merged**